onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,49 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
+/** Check if it will be possible to remove or fuse a node.
     We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+    - The node should not produce a graph output unless something else will produce that output.
+    - Only one of the outputs is used by downstream operators (multiple output edges are allowed).
+    - If the Node has a single incoming node, we can remove the Node and connect its incoming node to its 
+      outgoing nodes, if doing so does not clash with any values in any relevant subgraphs. 
+@param removing_output 
+       If true the output from the node will be removed and replaced with its input.
+       If false a new Node or initializer is being created that will produce output with the same name. In this case
+       it is safe to remove the node even if it provides a graph output.*/
```

Comment:
I think at this point, it is not clear why we care about this removing_output. Maybe it is simpler to say that if it is false, we dont care if node_output_is_in_graph_output; if true, we care. And explain what happens when we do the removal. We can also remove the first bullet comment, since this covers it.